### PR TITLE
air alarm signal ports and other stuff

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
- adds output ports to air alarm for each state, set to high when in it and low when not
- light air alarm refactoring
- hopefully made air alarm support working without AccessReader
- added support for sending packets to not normal networks, made device link InvokePort force all packets onto the Wireless network since obviously air alarm shouldnt send logic stuff to scrubbers and stuff, doesnt affect operation otherwise
- stuff

**Media**

an airlock controller that prevents the loss of air
air alarm is set to be at danger above 10kPa, warning at 100Pa, so normal when its completely empty
normal output -> not gate -> bolt input

https://github.com/space-wizards/space-station-14/assets/39013340/4362b432-d1f6-4d8e-8a89-9422d9435484

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- tweak: Air alarms now have output signal ports for each state.

### CODE REVIEW

**Comment by Partmedia** [comment]
Amazing, thank you for your contribution. Of course, separating "refactoring" commits from "change" commits makes this easier to review.

File: Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs [hunk 1]
```diff
@@ -219,14 +222,23 @@ private void OnClose(EntityUid uid, AirAlarmComponent component, BoundUIClosedEv
...
     private void OnShutdown(EntityUid uid, AirAlarmComponent component, ComponentShutdown args)
     {
         _activeUserInterfaces.Remove(uid);
     }
 
     private void OnInteract(EntityUid uid, AirAlarmComponent component, InteractHandEvent args)
     {
-        if (!_interactionSystem.InRangeUnobstructed(args.User, args.Target))
+        if (!_interaction.InRangeUnobstructed(args.User, args.Target))
             return;
```
Comment (metalgearsloth) [inline Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs:242]: This check should be unneeded as SharedInteractionSystem checks inRangeUnobstructed for this.

File: Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs [hunk 2]
```diff
@@ -219,14 +222,23 @@ private void OnClose(EntityUid uid, AirAlarmComponent component, BoundUIClosedEv
...
     }
 
+    private void OnInit(EntityUid uid, AirAlarmComponent comp, ComponentInit args)
+    {
+        _deviceLink.EnsureSourcePorts(uid, comp.DangerPort, comp.WarningPort, comp.NormalPort);
+
+        // for mapped linked air alarms, start with high so when it changes for the first time it goes from high to low
+        // without this the output would suddenly get sent a low signal after nothing which is bad
+        _deviceLink.SendSignal(uid, GetPort(comp), true);
+    }
```
Comment (metalgearsloth) [inline Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs:232]: Does this cause any issue if map is saved and loaded coz Init is run every time.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by metalgearsloth** [comment]
@juliangiebel

File: Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs (hunk 2 above)
Comment (deltanedas) [reply Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs:232]: should it be moved to MapInit then, and does MapInit run after ComponentInit

**Comment by juliangiebel** [comment]
I'll review this tomorrow

**Review by juliangiebel — APPROVED** [review]
Lgtm.

File: Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs (hunk 2 above)
Comment (mirrorcult) [reply Content.Server/Atmos/Monitor/Systems/AirAlarmSystem.cs:231]: yes

**Comment by deltanedas** [comment]
@metalgearsloth a

**Review by mirrorcult — APPROVED** [review]

**Comment by Myakot** [comment]
God bless you

**Comment by t1m1yep** [comment]
i love you

**Pull request merged**